discovery/gossiper.go

```diff
@@ -1469,23 +1469,34 @@ func (d *AuthenticatedGossiper) networkHandler(ctx context.Context) {
 			log.Tracef("Received channel %d policy update requests",
 				len(policyUpdate.edgesToUpdate))
 
-			// First, we'll now create new fully signed updates for
-			// the affected channels and also update the underlying
-			// graph with the new state.
-			newChanUpdates, err := d.processChanPolicyUpdate(
-				ctx, policyUpdate.edgesToUpdate,
-			)
-			policyUpdate.errChan <- err
-			if err != nil {
-				log.Errorf("Unable to craft policy updates: %v",
-					err)
-				continue
-			}
+			// Process policy updates asynchronously to avoid
+			// blocking the main network handler loop because the
+			// graph db has to be updated here which might be slow
+			// to acquire the db lock.
+			d.wg.Add(1)
```

Comment:
Perhaps we should still attempt to limit concurrency here? 

One thing I still don't understand is why in the reported case, the db lock was held for so long.

Reply:
Yeah also not sure why it held the lock for so long let's not change anything here for now until we get more data.